### ISSUE TITLE
make cmdstanr a suggested dependency

### ISSUE THREAD

**Issue by kyleam** [metrumresearchgroup/bbr.bayes#31]
Once nmbayes support lands (gh-30), consider making cmdstanr optional so that NONMEM users don't need to install a relatively heavy dependency.

---

https://github.com/metrumresearchgroup/bbr.bayes/pull/30#issuecomment-1302514238

> > make cmdstanr suggested package?
> 
> I think this is probably the right idea, but should definitely be a separate PR, because we'll have to carefully insert some `require(cmdstanr)` statements throughout the code, obviously. I'm hoping not too many places, but we'll see.

**Comment by seth127** [comment]
In reviewing #106 I thought about this again. 

> This issue can essentially be moved onto whatever milestone becomes the "nmbayes initial release."

It seems like we're getting close that milestone. Should we consider this now/soon?

Related to this, should we think about what the "easy path" is for users wanting to install `bbr.bayes` dependencies, considering that _most_ users will either care about using it for Stan _or_ for NONMEM (but likely not both)? 

Keeping stan/NONMEM-specific packages in `Suggests` makes sense, but then it's not clear to me how a user knows _which_ packages from `Suggests` they will want to add. Furthermore, most of our initial users will be installing via `pkgr`, which means they control this by specifying `Suggests: true` for `bbr.bayes`. However, this would install both the Stan-specific and NONMEM-specific packages, which is exactly what we were trying to avoid by adding them to `Suggests`.

This isn't a huge problem, but I'm also not seeing an obvious solution. We may have just painted ourself into a corner with the original package design here. I'm open to any thoughts on this. Also, if there is going to be much discussion on it, we may want to move it to a dedicated issue as well.

**Comment by kyleam** [comment]
> It seems like we're getting close that milestone. Should we consider this now/soon?

Sure (any time after gh-106 merges and before next release makes sense to me).

I initially stated the rationale as "cmdstanr optional so that NONMEM users don't need to install a relatively heavy dependency"... I don't think cmdstan**r** weighs too much, and (based on quick `pkgr inspect` scanning) it doesn't lead to any new dependencies being pulled in.  cmdstan, on the other hand, is very heavy, but that doesn't affect NONMEM users that pull cmdstanr in but don't use it.

Anyway, we may still want to do it regardless of whether it counts as "heavy", but it seems worth clarifying what our motivation is.

> Related to this, should we think about what the "easy path" is for users wanting to install `bbr.bayes` dependencies, considering that _most_ users will either care about using it for Stan _or_ for NONMEM (but likely not both)? [...]

I don't know.  I guess I just see it as something that comes along with defining something as an optional dependency.  It lets users that want to be more selective be so (with whatever installation tools they're using), knowing they may need to react to an error that tells them to install a suggested dependency.  And then those that don't want to worry about it can always just install everything, trading install time and space.

(In the Python world, you can group optional dependencies into named features via `extras_requires`, but I don't know of anything like that for R.)

**Comment by kyleam** [comment]
Based on offline chat, we're going to leave cmdstanr as is for the upcoming release.
